test/unit-test/rtcp_api/rtcp_api_utest.c

```diff
@@ -0,0 +1,1017 @@
+                                  &( rtcpPliPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    result = Rtcp_ParsePliPacket( &( context ),
+                                  &( rtcpPacket ),
+                                  NULL );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    rtcpPacket.pPayload = NULL;
+
+    result = Rtcp_ParsePliPacket( &( context ),
+                                  &( rtcpPacket ),
+                                  &( rtcpPliPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    rtcpPacket.payloadLength = 2;
+    payloadBuffer = ( uint8_t * ) malloc( rtcpPacket.payloadLength );
+    rtcpPacket.pPayload = payloadBuffer;
+
+    result = Rtcp_ParsePliPacket( &( context ),
+                                  &( rtcpPacket ),
+                                  &( rtcpPliPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+    free( payloadBuffer );
+
+    rtcpPacket.payloadLength = 10;
+    payloadBuffer = ( uint8_t * ) malloc( rtcpPacket.payloadLength );
+    rtcpPacket.pPayload = payloadBuffer;
+    rtcpPacket.header.packetType = RTCP_PACKET_UNKNOWN;
+
+    result = Rtcp_ParsePliPacket( &( context ),
+                                  &( rtcpPacket ),
+                                  &( rtcpPliPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+    free( payloadBuffer );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate RTCP Parse Sli Packet fail functionality for Bad Parameters.
+ */
+void test_rtcpParseSliPacket_BadParams( void )
+{
+    RtcpContext_t context;
+    RtcpPacket_t rtcpPacket = { 0 };
+    RtcpSliPacket_t rtcpSliPacket;
+    RtcpResult_t result;
+    uint8_t * payloadBuffer;
+
+    result = Rtcp_ParseSliPacket( NULL,
+                                  &( rtcpPacket ),
+                                  &( rtcpSliPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    result = Rtcp_ParseSliPacket( &( context ),
+                                  NULL,
+                                  &( rtcpSliPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    result = Rtcp_ParseSliPacket( &( context ),
+                                  &( rtcpPacket ),
+                                  NULL );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    rtcpPacket.pPayload = NULL;
+
+    result = Rtcp_ParseSliPacket( &( context ),
+                                  &( rtcpPacket ),
+                                  &( rtcpSliPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    rtcpPacket.payloadLength = 2;
+    payloadBuffer = ( uint8_t * ) malloc( rtcpPacket.payloadLength );
```

Comment:
Can it be replaced by static buffer? This test case might be fail if no buffer available and we don't want to handle that scenario.

Reply:
I can do that, so for all tests, no dynamic memory allocation should be given right, cause of no buffer available ? Please guide what to do